### PR TITLE
Fix clean validation envs workflow to use github context

### LINKED ISSUE #2965: Add clean validation workflow guidance to cicd doc
## Description

As a TRE Administrator
I want to have guidance on how to setup clean validation workflow in deployment repository.

### PR DESCRIPTION
**Pull request by LizaShak**
## What is being addressed

Fixes #2965 Add documentation on clean validation workflow.

### CODE REVIEW

**Review by tamirkamara — CHANGES_REQUESTED** [review]
@LizaShak Please add a fix for the cleanup script that will use github actions context variables for the repo owner/name (in this repo and the deployment one too).
I think `github.repository` the the right one to use.

File: .github/workflows/clean_validation_envs.yml [hunk 1]
```diff
@@ -29,6 +29,7 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           MAIN_TRE_ID: ${{ secrets.TRE_ID }}
+          GITHUB_REPOSITORY: ${{ github.repository}}
```
Comment (tamirkamara) [inline .github/workflows/clean_validation_envs.yml:32]: Can you try without this line?
Run the cleanup workflow on this branch to confirm

Comment (LizaShak) [reply .github/workflows/clean_validation_envs.yml:32]: Tried, it doesn't work without it

**Review by tamirkamara — APPROVED** [review]

**Pull request merged**